admin_organization_integration_test.go

```diff
@@ -238,39 +239,47 @@ func TestAdminOrganizations_Update(t *testing.T) {
 		assert.NoError(t, err)
 
 		assert.Equal(t, accessBetaTools, adminOrg.AccessBetaTools)
-		assert.Equal(t, globalModuleSharing, adminOrg.GlobalModuleSharing)
+		assert.Equal(t, adminOrg.GlobalModuleSharing, &globalModuleSharing)
 		assert.Equal(t, isDisabled, adminOrg.IsDisabled)
 		assert.Equal(t, terraformBuildWorkerApplyTimeout, adminOrg.TerraformBuildWorkerApplyTimeout)
 		assert.Equal(t, terraformBuildWorkerPlanTimeout, adminOrg.TerraformBuildWorkerPlanTimeout)
 		assert.Equal(t, terraformWorkerSudoEnabled, adminOrg.TerraformWorkerSudoEnabled)
+		assert.Nil(t, adminOrg.WorkspaceLimit, "default workspace limit should be nil")
 
 		isDisabled = true
 		globalModuleSharing = true
+		workspaceLimit := 42
 		opts = AdminOrganizationUpdateOptions{
 			GlobalModuleSharing: &globalModuleSharing,
 			IsDisabled:          &isDisabled,
+			WorkspaceLimit:      &workspaceLimit,
 		}
 
 		adminOrg, err = client.Admin.Organizations.Update(ctx, org.Name, opts)
 		assert.NoError(t, err)
 		assert.NotNilf(t, adminOrg, "Org returned as nil when it shouldn't be.")
 
-		assert.Equal(t, adminOrg.GlobalModuleSharing, globalModuleSharing)
+		assert.Equal(t, adminOrg.GlobalModuleSharing, &globalModuleSharing)
 		assert.Equal(t, adminOrg.IsDisabled, isDisabled)
+		assert.Equal(t, &workspaceLimit, adminOrg.WorkspaceLimit)
 
 		globalModuleSharing = false
 		isDisabled = false
+		workspaceLimit = 0
 		opts = AdminOrganizationUpdateOptions{
 			GlobalModuleSharing: &globalModuleSharing,
 			IsDisabled:          &isDisabled,
+			WorkspaceLimit:      &workspaceLimit,
 		}
 
 		adminOrg, err = client.Admin.Organizations.Update(ctx, org.Name, opts)
 		assert.NoError(t, err)
 		assert.NotNilf(t, adminOrg, "Org returned as nil when it shouldn't be.")
 
-		assert.Equal(t, adminOrg.GlobalModuleSharing, globalModuleSharing)
+		assert.Equal(t, &globalModuleSharing, adminOrg.GlobalModuleSharing)
```

Comment:
I'm a little confused by the argument swapping here and on L241. The other assertions seem to put the parameter second and the model first. And also by the address operator. What's going on?

Reply:
> It's worth noting that I had to make a few changes to the existing test (specifically, assertions about the GlobalModuleSharing flag) in order to get these tests to pass. Since those tests don't get run in CI, I'm thinking they probably haven't run in a long while.

OK ! Nevermind !